keepalived/check/ipwrapper.c

```diff
@@ -713,9 +713,51 @@ rs_exist(real_server_t * old_rs, list l)
 	return NULL;
 }
 
+static void
+migrate_failed_checkers(real_server_t *old_rs, real_server_t *new_rs)
+{
+	list l;
+	element e, e1;
+	checker_t *old_c, *new_c;
+	checker_id_t *id;
+
+	l = alloc_list(NULL, NULL);
+	for (e = LIST_HEAD(old_checkers_queue); e; ELEMENT_NEXT(e)) {
+		old_c = ELEMENT_DATA(e);
+		if (old_c->rs == old_rs) {
+			list_add(l, old_c);
+		}
+	}
+
+	if (LIST_ISEMPTY(l))
+		goto end;
+
+	for (e = LIST_HEAD(checkers_queue); e; ELEMENT_NEXT(e)) {
+		new_c = ELEMENT_DATA(e);
+		if (new_c->rs != new_rs || !new_c->compare)
```

Comment:
It's a consideration of a checker that does not set compare callback function.
If you don't allow such a checker, I think unnecessary too.